.eslintrc.json

```diff
@@ -30,10 +27,42 @@
     "testing-library/prefer-explicit-assert": "warn",
     "testing-library/prefer-presence-queries": "error",
     "testing-library/prefer-screen-queries": "error",
-    "testing-library/prefer-wait-for": "error",
-    "no-use-before-define": "off",
-    "@typescript-eslint/no-use-before-define": [
-      "error"
-    ]
-  }
+    "testing-library/prefer-wait-for": "error"
+  },
+  "overrides": [
+    {
+      "files": ["**/*.ts", "**/*.tsx"],
+      "env": {"browser": true, "es6": true, "node": true},
+      "extends": [
+        "eslint:recommended",
+        "plugin:@typescript-eslint/eslint-recommended",
+        "plugin:@typescript-eslint/recommended"
+      ],
+      "globals": {"Atomics": "readonly", "SharedArrayBuffer": "readonly"},
+      "parser": "@typescript-eslint/parser",
+      "parserOptions": {
+        "ecmaFeatures": {"jsx": true},
+        "ecmaVersion": 2018,
+        "sourceType": "module",
+        "project": "./tsconfig.json"
+      },
+      "plugins": ["@typescript-eslint"],
+      "rules": {
+        "no-useless-computed-key": "off",
```

Comment:
I don't really know what I'm doing here. Copied all of the rules from above because it sounded like a good idea? The rest of this overrides section came from stackoverflow. @codyarose @SeanGroff please look this over =]

Reply:
I believe you want to copy rules that are TypeScript related here.
I think we need to move them all back accept 
```
"no-use-before-define": "off",
 "@typescript-eslint/no-use-before-define": ["error"]
 ```